File: createSitemap.php
<HTML><HEAD>
<META content="text/html; charset=utf-8" http-equiv=Content-Type>
<META name=GENERATOR content="MSHTML 9.00.8112.16437"></HEAD>
<BODY>

<?php

setlocale(LC_ALL, "ru_RU");

require_once("1cengine/site/secrets.php");

my_dbConnect();

$filename = 'sitemapBase.xml';
$handle = fopen($filename, 'r');
$size = filesize ( 'sitemapBase.xml' );
    // считываем весь файл в переменную $content
$sitemap = fread ( $handle, $size );
    //  закрываем процесс
fclose ( $handle ); 

$filename = 'sitemap.xml';
$handle = fopen($filename, 'w');
fwrite($handle, $sitemap);

$r = mysql_query("SELECT `offers`.`display_name`, `offers`.`char_name` FROM `offers`");

if (mysql_num_rows($r)>0){
    while($row = mysql_fetch_array($r, MYSQL_NUM)){
        
        $addRow = '<url>
                  <loc>http://trimet.ru/1cengine/site/?ref='.rawurlencode($row[0].' '.$row[1]).'</loc>
                  <priority>0.7</priority>
                  <changefreq>daily</changefreq>
               </url>
               ';

        fwrite($handle, $addRow);

    }
}

fwrite($handle, '</urlset>');

fclose($handle);

echo 'Sitemap generated';
// print_r($groupArrays[1]['_c']['m:Группа'][0]['_c']['m:Группа'][0]['_c']['m:Предмет']); 

?>

</BODY></HTML>

File: createSitemap.py
#!/usr/bin/python2.6
# -*- coding: utf-8 -*-


import sys
import os
import cgitb
cgitb.enable()
import urllib

print("Content-Type: text/html; charset=utf-8\n")

lib_path = os.path.abspath('1cengine/site/')
sys.path.append(lib_path)

from secrets import *


def getItems():
    connector = myDBC("catalog")
    connector.dbConnect()
    row = connector.dbExecute("""
           SELECT `item`.`name` , `char`.`name`
            FROM `item` , `char`
            WHERE `char`.`item_ref` = `item`.`id`
            ORDER BY `item`.`name` , `char`.`name`
        """)

    connector.dbClose()

    return row


def writeItemsToFile(row):

    addRow = ""
    for x in row:

        addRow = addRow + """
            <url>\n
                <loc>
                    http://trimet.ru/1cengine/site/?ref="""\
                    + urllib.quote(x[0] + " " + x[1]) + \
            """
                </loc>\n
                <priority>0.7</priority>\n
                <changefreq>daily</changefreq>\n
           </url>\n
           """
        # print addRow

    addRow = addRow + '</urlset>'

    _PATH = os.path.abspath(os.path.dirname(__file__))

    sitemapBaseFilepath = os.path.join(_PATH, 'sitemapBase.xml')
    sitemapFilepath = os.path.join(_PATH, 'sitemap.xml')

    sitemapBase = open(sitemapBaseFilepath, 'r')
    sitemapBase_string = sitemapBase.read()
    sitemapBase.close()

    sitemapFile = open(sitemapFilepath, 'w+')
    sitemap = sitemapBase_string + addRow
    sitemapFile.write(sitemap)
    sitemapFile.close()

    print "sitemap generated"

row = getItems()
writeItemsToFile(row)
